fuel-p2p/src/orchestrator.rs

```diff
@@ -20,58 +19,58 @@ use crate::{
     service::{FuelP2PEvent, FuelP2PService},
 };
 
-type ResponseFuture = Pin<Box<dyn Future<Output = Option<(OutboundResponse, RequestId)>>>>;
-
 pub struct NetworkOrchestrator {
-    p2p_service: FuelP2PService,
+    p2p_config: P2PConfig,
 
     /// receives messages from different Fuel components
     rx_request_event: Receiver<P2pRequestEvent>,
+    rx_outbound_responses: Receiver<Option<(OutboundResponse, RequestId)>>,
 
     // senders
     tx_consensus: Sender<ConsensusBroadcast>,
     tx_transaction: Sender<TransactionBroadcast>,
     tx_block: Sender<BlockBroadcast>,
+    tx_outbound_responses: Sender<Option<(OutboundResponse, RequestId)>>,
 
-    db: Arc<dyn RelayerDb>,
-
-    outbound_responses: FuturesUnordered<ResponseFuture>,
+    db: Arc<Box<dyn P2pDb>>,
 }
 
 impl NetworkOrchestrator {
-    pub async fn new(
-        local_keypair: Keypair,
+    pub fn new(
         p2p_config: P2PConfig,
         rx_request_event: Receiver<P2pRequestEvent>,
 
         tx_consensus: Sender<ConsensusBroadcast>,
         tx_transaction: Sender<TransactionBroadcast>,
         tx_block: Sender<BlockBroadcast>,
 
-        db: Arc<dyn RelayerDb>,
-    ) -> Result<Self, Box<dyn Error>> {
-        let p2p_service = FuelP2PService::new(local_keypair, p2p_config).await?;
+        db: Arc<Box<dyn P2pDb>>,
+    ) -> Self {
+        let (tx_outbound_responses, rx_outbound_responses) = tokio::sync::mpsc::channel(100);
 
-        Ok(Self {
-            p2p_service,
+        Self {
+            p2p_config,
             rx_request_event,
+            rx_outbound_responses,
             tx_block,
             tx_consensus,
             tx_transaction,
+            tx_outbound_responses,
             db,
-            outbound_responses: Default::default(),
-        })
+        }
     }
 
-    pub async fn run(&mut self) {
+    pub async fn run(mut self) -> anyhow::Result<Self> {
+        let mut p2p_service = FuelP2PService::new(self.p2p_config.clone()).await?;
```

Comment:
We should have a bypass for running this service if p2p networking is disabled / not configured. fuel-core is heavily utilized via SDK integration tests for contract development, and we don't want to increase the cost of localized fuel-core testing with lots of libp2p connection setup and teardown.

Reply:
good point, I think this should be somewhere where [this service is started](https://github.com/FuelLabs/fuel-core/pull/492/files#diff-9687cbe0598079f7848f4f130bdb462da74313456432c93874f3e8473fcda142R116) instead of the `run()` method itself.

Maybe having different `start_modules()` methods, like `start_with_network()` and without ?

Reply:
I think it should be up to each module to decide whether it's going to run based on the module related config, rather than branching at the top level. So in this case, I think p2p should not run by default unless things like peering keys and bootstrap nodes are setup.

Reply:
I don't disagree, but we already have [conditional module starting](https://github.com/FuelLabs/fuel-core/blob/46e952895ee32521f69262c5992af137ddba40a5/fuel-core/src/service/modules.rs#L97) so I figured we add another one for the network, instead of having a `start()` method on a module which would not always start actually.